src/osd/PGLog.h

```diff
@@ -512,11 +512,6 @@ struct PGLog {
     if (missing.is_missing(oid, v)) {
       missing.got(oid, v);
       
-      // raise last_complete?
-      if (missing.missing.empty()) {
-	log.complete_to = log.log.end();
-	info.last_complete = info.last_update;
-      }
       while (log.complete_to != log.log.end()) {
 	if (missing.missing[missing.rmissing.begin()->second].need <=
 	    log.complete_to->version)
```

Comment:
Good catch! Yeah.. I will update the commit.

Other than other, does the analysis make sense for this problem?